src/workerd/jsg/limits.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <kj/common.h>
+
+namespace workerd::jsg {
+
+class IsolateLimitEnforcer {
```

Comment:
Can we instead devise a way to get at the `Worker::Isolate` object from a `jsg::Lock`? Then you could grab the `IsolateLimitEnforcer` from there, as well as other useful things. It seems weird to introduce the "limit enforcer" concept spjcifically into JSG.

Of course, we don't want to introduce a cyclic dependency on `Worker::Isolate` so I guess we need some mechanism to store an arbitrary `void*` which we say is reserved for this, and then we can have a function like `static Worker::Isolate& Worker::Isolate::from(jsg::Lock&)` to grab the isolate reference in a type-safe way.

Reply:
Ok, updated. Eliminated the jsg IsolateLimitEnforcer addition and added the `Worker::Isolate::from(...)` mechanism.